### PR TITLE
chore: adds type checking and fixes existing type errors

### PR DESCRIPTION
**Pull request by snebjorn**
Adds `yarn typecheck` to run type checking in packages. Also added it to the PR Check.

100% vibed the type fixes. So no idea if they make sense.

## Summary by CodeRabbit

* **Chores**
  * Added TypeScript type checking (via `tsc --noEmit`) across all packages as a development script.
  * Integrated typecheck step into PR checks workflow and build documentation.

* **Bug Fixes**
  * Improved null/undefined handling for API metadata, ensuring fallbacks when titles are missing.
  * Enhanced error messaging for unresolved references.
  * Fixed parameter default value serialization.

* **Tests**
  * Added new test utilities for consistent context setup.
  * Expanded test coverage for edge cases including nullable objects and null defaults.

### CODE REVIEW

File: packages/mock/src/faker/getters/object.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      operationId: 'getNullableObject',
+      tags: [],
+      context,
+      imports: [],
+      existingReferencedProperties: [],
+      splitMockImplementations: [],
+    });
+
+    expect(result.value).toBe('faker.helpers.arrayElement([null,])');
+  });
```
[the remark this answers is not in the document]
Comment (snebjorn) [reply packages/mock/src/faker/getters/object.test.ts:30]: You're most likely correct about this. But this PR isn't trying to change behavior. It's only trying to fix type errors without changing the generated output. Fixing this will change the output.

Can you scan the open issues and see if any cover this bug?

[the remark this answers is not in the document]
Comment (snebjorn) [reply packages/mock/src/faker/getters/object.test.ts:30]: please do
